src/elasticsearch_cleanup/main.py

```diff
@@ -0,0 +1,200 @@
+"""This script is used for removing duplicate documents from an OpenSearch index.
+
+The main logic includes:
+1. Setting up the OpenSearch client.
+2. Fetching all documents from the specified OpenSearch index.
+3. Identifying duplicate documents based on a comparison field.
+4. Removing the duplicates if not in dry-run mode or logging the bulk actions if in dry-run mode.
+"""
+
+import json
+import logging
+from itertools import tee
+from typing import Any, Generator
+
+from opensearchpy import OpenSearch, helpers
+
+from elasticsearch_cleanup import config, consts
+from elasticsearch_cleanup.logger import get_logger
+from elasticsearch_cleanup.utils import str_to_bool
+
+logger = get_logger(config.LOG_LEVEL)
+
+
+def remove_documents(
+    opensearch_client: OpenSearch,
+    actions: list[dict[str, str]],
+    index: str,
+    logger: logging.Logger,
+):
+    """Removes specified documents from the OpenSearch index.
+
+    Args:
+        opensearch_client: The client to communicate with OpenSearch.
+        actions: List of actions to remove documents.
+        index: The name of the index.
+        logger: Logger for logging info and exceptions.
+    """
+    bulk_errors = helpers.bulk(client=opensearch_client, actions=actions)
+    logger.info(f"Removed '{len(actions)}' duplicates from '{index}'.")
+
+    opensearch_client.indices.refresh(index=index)
+
+    if len(bulk_errors) > 0:
+        logger.exception("failed to bulk some actions:")
+        for err in bulk_errors:
+            logger.exception(str(err))
```

Comment:
It's a reoccurring thing throughout our code base, but..
It's better to crash as hard as we can for almost any kind of exception we get. Logging the exception masks the error and exposes us to problems in the future where things go south